src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1281,6 +1304,186 @@ internal BoundExpression BindCollectionBuilderCreate(
             return collectionCreation;
         }
 
+        private BoundExpression BindInterfaceTargetCollectionMethodGroup(
+            SyntaxNode syntax,
+            CollectionExpressionTypeKind collectionTypeKind,
+            NamedTypeSymbol targetType,
+            BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(ContainingType is { });
+            const string methodName = "<signature>"; // Arbitrary name.
+            var builder = ArrayBuilder<MethodSymbol>.GetInstance();
+            addSignatures(builder, syntax, ContainingType, methodName, targetType, diagnostics);
+            if (builder.IsEmpty)
+            {
+                return new BoundBadExpression(syntax, LookupResultKind.Empty, symbols: [], childBoundNodes: [], type: CreateErrorType());
+            }
+            return new BoundMethodGroup(
+                syntax,
+                typeArgumentsOpt: default,
+                methodName,
+                methods: builder.ToImmutableAndFree(),
+                lookupSymbolOpt: null,
+                lookupError: null,
+                flags: BoundMethodGroupFlags.None,
+                functionType: null,
+                receiverOpt: null,
+                resultKind: LookupResultKind.Viable);
+
+            void addSignatures(
```

Comment:
This local function captures `collectionTypeKind`. Consider passing that explicitly to reduce allocation

Reply:
Good catch, I was only intending to capture `this`. I've changed the local function to an instance method to enforce that.